packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for invoking an API Endpoint with ApiGatewayInvoke
+ */
+export interface ApiGatewayInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * hostname of an API Gateway URL
+   * @example {ApiId}.execute-api.{region}.amazonaws.com
+   */
+  readonly apiEndpoint: string;
```

Comment:
how about 

```ts
readonly api: apigateway.IRestAPI;
```

And maybe generate the `apiEndpoint`  like this?

```ts
const apiStack = cdk.Stack.of(props.api);
this.apiEndpoint = ${props.api.restApiId}.execute-api.${apiStack.region}.${apiStack.urlSuffix}
```

Reply:
this suggestion is essentially what I had in mind too

Reply:
Implementing now. this cleans it up nicely, thanks for the suggestion.